edge-hub/src/Microsoft.Azure.Devices.Routing.Core/Router.cs

```diff
@@ -130,14 +130,13 @@ public async Task RemoveRoute(string id)
             {
                 this.CheckClosed();
                 ImmutableDictionary<string, Route> snapshot = this.routes;
-                ImmutableHashSet<Endpoint> currentEndpoints = this.Routes.SelectMany(r => r.Endpoints).ToImmutableHashSet();
-                this.routes.Value = snapshot.Remove(id);
-                ISet<string> removedEndpointIds = currentEndpoints.Except(this.Routes.SelectMany(r => r.Endpoints))
-                    .Select(e => e.Id)
-                    .ToImmutableHashSet();
-
-                this.evaluator.RemoveRoute(id);
-                await this.dispatcher.RemoveEndpoints(removedEndpointIds);
+
+                if (snapshot.TryGetValue(id, out Route removedRoute))

```

Comment:
we have `TryRemove` method to make 1 method call intead of 2.

Reply:
I need a ref to removedRoute to get the endpoint id, which is different then the route id being passed in.

---
In reply to: [376152598](https://github.com/Azure/iotedge/pull/2469#discussion_r376152598) [](ancestors = 376152598)